lib/src/main/java/io/ably/lib/types/BaseMessage.java

```diff
@@ -93,9 +98,13 @@ public void decode(ChannelOptions opts) throws AblyException {
 						catch(JsonParseException e) { throw AblyException.fromThrowable(e); }
 						continue;
 					}
-					if(xform == "cipher" && opts != null && opts.encrypted) {
-						data = opts.getCipher().decrypt((byte[])data);
-						continue;
+					if(xform == "cipher") {
+						if (opts != null && opts.encrypted) {
+							data = opts.getCipher().decrypt((byte[]) data);
+							continue;
+						}
+						else
+							Log.e(TAG, "Encrypted message received but encryption is not set up");
```

Comment:
I'm not sure this is an error - I would make this `Log.i` I think.

Also lets say ".. encryption is not enabled on channel", and also include the channel name in the message pls.